### PR TITLE
Add hiscores command

### PR DESCRIPTION
**Pull request by DarkMatterMatt**
Usage: `!hiscores`, extra arguments are ignored
Alias: `!highscores`, `!leaderboards`

Data is sourced from the normalised data table

There's a constant `const maxHiscoresLength = 10`, I don't know where it should be put (some config is in files, some things like `APPRECIATIONS` are in the code, some things like `milestoneStartDate` are put in the spreadsheet.

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ const util = require('util');
 const fs = require('fs');
 const readFile = util.promisify(fs.readFile);
 
+const AsciiTable = require('ascii-table')
```
Comment (ErnWong) [inline main.js]: Missing semicolon.

```suggestion
const AsciiTable = require('ascii-table');
```

File: main.js [hunk 2]
```diff
@@ -101,6 +102,24 @@ async function init() {
           message.reply('Could not log hours :( yell at ernest');
         }
         break;
+
+      case '!hiscores':
+      case '!highscores':
+      case '!leaderboards':
+        const totalHours = await getTotalHours(googleClient);
+        const maxHiscoresLength = 10;
```
Comment (ErnWong) [inline main.js]: Let's use uppercase for these kinds of constant parameters.

```suggestion
        const MAX_HISCORES_LENGTH = 10;
```

File: main.js [hunk 3]
```diff
@@ -101,6 +102,24 @@ async function init() {
...
+
+        // sort people by hours, cut list length to maxHiscoresLength
+        const hiscores = [];
+        Object.keys(totalHours).forEach(name => hiscores.push([name, totalHours[name]]));
+        hiscores.sort((a, b) => b[1] - a[1]);
+        hiscores.splice(maxHiscoresLength);
```
Comment (ErnWong) [inline main.js]:
```suggestion
        hiscores.splice(MAX_HISCORES_LENGTH);
```

File: main.js (hunk 2 above)
Comment (ErnWong) [reply main.js]: >  I don't know where it should be put

Yup, putting it here will do for now.

The three places for constants are:
- `config.json` are settings associated with the environment you're running on. E.g., my development environment will be using a different discord bot token compared with your development environment, both of which will be different to the production environment which will be using the actual discord bot's token.
- Google Sheets settings page: things that are likely to change, such as the milestone values. Putting them in the spreadsheet makes it easy for non-tech-savvy users to change.
- Inline with the code: things that probably won't change between environments and probably won't change over time. We can always move constants from here to the Google Sheets if we feel like it.

**Review by ErnWong — CHANGES_REQUESTED** [review]
Thanks! Works pretty well.

File: main.js (hunk 2 above)
Comment (DarkMatterMatt) [reply main.js]: Yeah, forgot to change it from lowercase from when it was a function argument

**Review by ErnWong — APPROVED** [review]

**Pull request merged**